lib/index.js

```diff
@@ -738,13 +738,13 @@ var Client = module.exports = function (config) {
           callCallback(err)
         })
         res.on('end', function () {
-          if (res.statusCode >= 301 && res.statusCode <= 307) {
+          if (res.statusCode !== 304 && res.statusCode >= 301 && res.statusCode <= 307) {
             options.path = Url.parse(res.headers.location, true).path
             httpSendRequest()
             return
           }
 
-          if (res.statusCode >= 400 || res.statusCode < 10) {
+          if (res.statusCode === 304 || res.statusCode >= 400 || res.statusCode < 10) {
             callCallback(new error.HttpError(data, res.statusCode, res.headers))
```

Comment:
Can you add a `304` status in https://github.com/octokit/node-github/blob/master/lib/error.js#L47-L48? Maybe add a comment to reference this pull request, as it looks a little confusing to have a single `304` besides all the 400+ error codes

Reply:
sure thing, i’ll try to do it tomorrow

Reply:
i was thinking, along with more comments, I could make the 301-307 if statement predicate on the existence of a truthy location header instead of calling out 304... the in the exception case if statement switching it to be all status’s outside the 200-299 range... thoughts?

Reply:
I've added the 304 to that file